singularity/code/savegame.py

```diff
@@ -687,6 +688,36 @@ def savegame_exists(savegame_name):
 
     return True
 
+def sanitize_filename(filename):
```

Comment:
I think I am more inclined to let the user know that the file name is not valid rather than change it behind their back in a "lossy way" (with a few exceptions).  As an example:

 * Stripping leading/trailing whitespace: OK
 * Replacing a leading dot or dash with underscore: Prefer a "Not valid" error
 * Funny characters like <>: etc.: Prefer a "Not valid error" (or encoded in a reversible way)
 * Space to underscore: OKish (I would prefer underscores were forbidden or encoded in a reversible way; but I would be fine with it)
 * Reserved path names on Windows: OK 
 * Length: An error rather than a silent truncation.  Note, I think the need the file system encoding before we check the length as `len(name) != len(name.encode('<fs-encoding>'))`.

(Obviously, this is great and all, but I am not sure we have a good setup for reporting the error in the first place, which may be an Achilles heel in my review)

Reply:
Does the UI code allow for changing the color of a single character? We could highlight the forbidden characters then.

Alternatively, list forbidden characters in a text field.

It would also be good to block the "OK" button as long as the filename isn't valid.

Reply:
We can block the "OK" button by setting its `.enabled` to False.  We might be able to change the background color of the entire text field but not a single character nor does the UI code have any fancy pop-ups.

Reply:
Mostly done:

- OK button gets blocked on illegal filename
- User sees an error message in the dialog when it's blocked
- I dropped the idea of replacing `_` with ` `

Filename length check is not correct yet